### PR TITLE
Add check for path.Join() to pre commit hook

### PR DESCRIPTION
**Pull request by zanderz**
@maxtaco @oconnor663

### CODE REVIEW

File: git-hooks/pre-commit [hunk 1]
```diff
@@ -74,8 +74,26 @@ check_make_lint() {
...
 }
 
+#
+# Don't allow path.Join()
+#
+check_path_Join()
+{
+    for file in $(git diff --cached --name-only --diff-filter=ACM | grep '.go$' | grep -v go/vendor/ | grep -v vendor/)
+    do
+    pathjoins=$(grep -E ' path\.Join\(')
```
Comment (maxtaco) [inline git-hooks/pre-commit]: you might not always have a space before it. Does `'\bpath\.Join\('` work instead?  `\b` means "word boundary"

**Comment by zanderz** [comment]
You were right, of course. Not only that, but my hook was wrong, I submitted too hastily, but I think this will do it.

**Comment by jacobhaven** [comment]
I mean, it's still possible to use `path.Join` to good effect (mainly for URL paths)

**Comment by zanderz** [comment]
whoops, good point. Maybe this isn't the way to go.

**Comment by maxtaco** [comment]
It's a good point, but on the other hand, it's just a precommit hook (and not a C/I failure). You can always disable it to push a change through.

**Comment by oconnor663** [comment]
LTGM, especially since this only applies to changes you're making and not to the whole repo. We can remove it later if it turns out to be inconvenient?

**Pull request merged**